### DISCUSSION TITLE
Error when comparing github.actor to string in if statement

### DISCUSSION THREAD

**Discussion by danielinclouds** [community/community#27015] [Actions]
<p>Hey,</p>
<p>I’m getting an error when trying to compare github.actor to my username in the if statement.</p>
<p>This is the error I’m getting:</p>
<pre><code>The workflow is not valid. .github/workflows/compare.yml (Line: 18, Col: 11): Unexpected symbol: '"my_username"'. Located at position 17 within expression: github.actor == "my_username"
</code></pre>
<p>My goal is to:</p>
<p>* only run action if my user pushed the code</p>
<pre><code>name: Compare user to actor

on:
  push:

jobs:

  build:
    name: Compare user to actor
    runs-on: ubuntu-latest
    steps:

    - name: Check out code
      uses: actions/checkout@v2

    - name: Actor
      if: github.actor == "my_username"
      env:
        ACTOR: ${{ github.actor }}
        PUSHER: ${{ github.event.pusher.name }}
      run: |
        echo "$ACTOR"
        echo "$PUSHER"
</code></pre>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/danielinclouds">@danielinclouds</a> ,</p>
<p>The error is due to double quotes are used for username, please use single quote, code as below:</p>
<pre><code>- name: Actor
        if: github.actor ==&lt;font color="#FF0000"&gt;'&lt;/font&gt;weide-zhou&lt;font color="#FF0000"&gt;' &lt;font color="#000000"&gt;# use '' to enclose username&lt;/font&gt;&lt;/font&gt; env:
          ACTOR: ${{ github.actor }}
          PUSHER: ${{ github.event.pusher.name }}
        run: |
          echo "$ACTOR"
          echo "$PUSHER"
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181101161-b52179a4-cb92-442b-82df-ead04321670a.png" alt="singalquote.png" data-base62-sha1="odVTBdcrJKFr7wcjukPO8ihekcn" width="228" height="189"></p>
<p>Thanks.</p>

**Comment by danielinclouds** [comment]
<p>Thanks, that fixed my problem.</p>
